forui/lib/src/widgets/rating/rating.dart

```diff
@@ -0,0 +1,183 @@
+      ..add(DoubleProperty('spacing', spacing));
+  }
+}
+
+class _FRatingState extends State<FRating> {
+  double _hoverValue = 0.0;
+  bool _isHovering = false;
+
+  double _calculateRating(double dx, double totalWidth) {
+    if (totalWidth <= 0) {
+      return 0;
+    }
+
+    final itemWidth = totalWidth / widget.count;
+    final x = dx.clamp(0.0, totalWidth);
+    final index = x ~/ itemWidth;
+    final localPosition = x - (index * itemWidth);
+
+    double rating = index + 1.0;
+    if (widget.halfFilledIcon != null && localPosition < itemWidth / 2) {
+      rating -= 0.5;
+    }
+
+    return rating.clamp(0.0, widget.count.toDouble());
+  }
+
+  void _handleHover(PointerHoverEvent event) {
+    final box = context.findRenderObject() as RenderBox?;
+    if (box == null) {
+      return;
+    }
+
+    final localPosition = box.globalToLocal(event.position);
+    setState(() => _hoverValue = _calculateRating(localPosition.dx, box.size.width));
+  }
+
+  void _handleTap(TapUpDetails details) {
+    final box = context.findRenderObject()! as RenderBox;
+    final localPosition = box.globalToLocal(details.globalPosition);
+    final rating = _calculateRating(localPosition.dx, box.size.width);
+
+    widget.onStateChanged?.call(rating);
+    setState(() => _isHovering = false);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = context.theme;
+    final style = widget.style ?? const FRatingStyle();
+    final effectiveEnabled = widget.enabled && widget.onStateChanged != null;
+    final effectiveValue = _isHovering ? _hoverValue : widget.value;
+
+    return MouseRegion(
+      onEnter: effectiveEnabled ? (_) => setState(() => _isHovering = true) : null,
+      onExit: effectiveEnabled ? (_) => setState(() => _isHovering = false) : null,
+      onHover: effectiveEnabled ? _handleHover : null,
+      child: GestureDetector(
+        onTapUp: effectiveEnabled ? _handleTap : null,
+        child: _RatingContent(
+          value: effectiveValue,
+          count: widget.count,
+          spacing: widget.spacing,
+          filledIcon: widget.filledIcon,
+          emptyIcon: widget.emptyIcon,
+          halfFilledIcon: widget.halfFilledIcon,
+          style: style,
+          theme: theme,
+          semanticsLabel: widget.semanticsLabel,
+          focusNode: widget.focusNode,
+          autofocus: widget.autofocus,
+          onFocusChange: widget.onFocusChange,
+        ),
+      ),
+    );
+  }
+}
+
+/// Internal widget to display the rating content
+
+/// Defines the visual properties for [FRating].
+///
+/// See also:
+/// * [FRating], which uses this class for its visual styling.
+final class FRatingStyle with Diagnosticable, _$FRatingStyleFunctions {
+  /// The color of the rating icons.
+  @override
+  final Color? color;
+
+  /// The size of the rating icons.
+  @override
+  final double? size;
+
+  /// Creates a [FRatingStyle].
+  const FRatingStyle({this.color, this.size});
+}
```

Comment:
I think these properties can be merged into a `IconThemeData`. `IconThemeData` can then be wrapped inside a `FWidgetStateMap` to account for the various states, i.e. selected & unselected.

I think you can reference `FTileContentStyle`.